meson.build

```diff
@@ -0,0 +1,362 @@
+
+cdata = configuration_data()
+cdata.set('_GNU_SOURCE', 1)
+cdata.set('FLATPAK_COMPILATION', 1)
+cdata.set('PACKAGE_MAJOR_VERSION', flatpak_major_version)
+cdata.set('PACKAGE_MINOR_VERSION', flatpak_minor_version)
+cdata.set('PACKAGE_MICRO_VERSION', flatpak_micro_version)
+cdata.set('PACKAGE_EXTRA_VERSION', flatpak_extra_version)
+cdata.set_quoted('PACKAGE_STRING', meson.project_name())
+cdata.set_quoted('PACKAGE_VERSION', meson.project_version())
+cdata.set_quoted('FLATPAK_BINDIR', get_option('prefix') / get_option('bindir'))
+cdata.set_quoted('FLATPAK_SYSTEMDIR', get_option('prefix') / system_install_dir)
+cdata.set_quoted('FLATPAK_CONFIGDIR', get_option('prefix') / get_option('sysconfdir') / 'flatpak')
+cdata.set_quoted('FLATPAK_BASEDIR', get_option('prefix') / get_option('datadir') / 'flatpak')
+cdata.set_quoted('FLATPAK_TRIGGERDIR', get_option('prefix') / get_option('datadir') / 'flatpak' / 'triggers')
+cdata.set_quoted('LIBEXECDIR', get_option('prefix') / get_option('libexecdir'))
+cdata.set_quoted('DATADIR', get_option('prefix') / get_option('datadir'))
+cdata.set_quoted('LOCALEDIR', get_option('prefix') / get_option('localedir'))
+cdata.set_quoted('SYSTEM_FONTS_DIR', get_option('system_fonts_dir'))
+cdata.set_quoted('SYSTEM_HELPER_USER', get_option('system_helper_user'))
+cdata.set_quoted('SYSTEM_FONT_CACHE_DIRS', ':'.join(get_option('system_font_cache_dirs')))
+cdata.set_quoted('G_LOG_DOMAIN', 'flatpak')
+cdata.set_quoted('GETTEXT_PACKAGE', 'flatpak')
+
+if get_option('system_bubblewrap') == ''
+  cdata.set_quoted('HELPER', get_option('prefix') / get_option('libexecdir') / 'flatpak-bwrap')
+else
+  cdata.set_quoted('HELPER', get_option('system_bubblewrap'))
+endif
+
+if get_option('system_dbus_proxy') == ''
+  cdata.set_quoted('DBUSPROXY', get_option('prefix') / get_option('libexecdir') / 'flatpak-dbus-proxy')
+else
+  cdata.set_quoted('DBUSPROXY', get_option('system_dbus_proxy'))
+endif
+
+# Flatpak is Linux-specific, so for now we assume that -fvisibility=hidden
+# is always supported
+cdata.set('FLATPAK_EXTERN', '__attribute__((visibility("default"))) extern')
+
+if glib_dep.version().version_compare('>=2.60')
+  # Ignore massive GTimeVal deprecation warnings in 2.62
+  cdata.set('GLIB_VERSION_MIN_REQUIRED', 'GLIB_VERSION_2_60')
+endif
+
+if appstream_dep.version().version_compare('>=0.14.0')
+  cdata.set('HAVE_APPSTREAM_0_14_0', 1)
+endif
+
+if libseccomp_dep.found()
+  cdata.set('ENABLE_SECCOMP', 1)
+endif
+
+if xau_dep.found()
+  cdata.set('ENABLE_XAUTH', 1)
+endif
+
+if not get_option('sandboxed_triggers')
+  cdata.set('DISABLE_SANDBOXED_TRIGGERS', 1)
+endif
+
+if cc.has_function(
+  'archive_read_support_filter_all',
+  dependencies : libarchive_dep,
+  prefix : '#include <archive.h>',
+)
+  cdata.set('HAVE_ARCHIVE_READ_SUPPORT_FILTER_ALL', 1)
+endif
+
+configure_file(
+  output : 'config.h',
+  configuration : cdata,
+)
+
+if meson.version().version_compare('>=0.53.0')
+  summary({
```

Comment:
You might also be interested in the `bool_yn` kwarg here, as that pretty-prints "true" / "false" into green YES or red NO.

Reply:
Yeah, probably good to have.